ci/hooks/check-on-stop.py

```
@@ -124,64 +265,161 @@ def should_skip_hook() -> bool:
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            text=True,
-            encoding="utf-8",
-            errors="replace",
-            cwd=str(PROJECT_ROOT),
-        )
-        test_proc_holder.append(proc)
+        t0 = time.perf_counter()
+        try:
+            proc = subprocess.Popen(
+                ["uv", "run", "test.py", "--cpp"],
```

Ruff (0.15.15)

[error] Starting a process with a partial executable path

(S607)